### ISSUE TITLE
phrase/2 malfunction

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#607]
v1.33.9
```
?- phrase((..., [L], ...), [a,b,c]).
   L = a
;  L = b
;  L = c
;  false.

?- phrase((..., [L], ...), Ls).
   Ls = [L]
;  Ls = [L,_A]
; ^C   % hangs or loops, hence ^C
Action or (h)elp: e
 Ls = [L,_A,_A|]
;

Sometimes Trealla continues as shown above with next answer which is an incomplete list: [L,_A,_A|]
or (e)nds as expected with false.
```

**Comment by infradig** [comment]
~/trealla (devel) $ tpl -O0
?- phrase((..., [L], ...), Ls).
   Ls = [L]
;  Ls = [L,_A]
;  Ls = [L,_A,_B]
;  Ls = [L,_A,_B,_C]
;  Ls = [L,_A,_B,_C,_D]
...

So probably some recent optimization change.

**Comment by flexoron** [comment]
Aha, Jesus, too fast...Did not perceive -O0 as a flag and thought you mean the CFLAG (-O3) in the Makefile.
Anyway, fixed with v1.33.12. Top!
